core/src/main/scala/tofu/syntax/bracket.scala

```diff
@@ -12,32 +12,117 @@ import tofu.{Finally, Guarantee}
 
 object bracket {
   implicit final class TofuBracketOps[F[_], A](private val fa: F[A]) extends AnyVal {
+
+    /**
+      * Apply function to [[fa]] with effectful transformation. In case of error [[fa]] is released
+      * @param use function to modify value contained in [[fa]]
+      * @param release function to release value contained in [[fa]]
+      * @return `F[B]` updated value
+      */
     def bracketIncomplete[B, C](
         use: A => F[B]
     )(release: A => F[C])(implicit F: Applicative[F], FG: Guarantee[F]): F[B]                     =
       FG.bracket(fa)(use) { case (a, success) => release(a).whenA(!success) }
 
+    /**
+      * Apply function to [[fa]] with effectful transformation. [[fa]] is always released
+      * @param use function to modify value contained in [[fa]]
+      * @param release function to release value contained in [[fa]]
+      * @return `F[B]` updated value
+      */
     def bracketAlways[B, C](
         use: A => F[B]
-    )(release: A => F[C])(FG: Guarantee[F]): F[B]                                                 =
+    )(release: A => F[C])(implicit FG: Guarantee[F]): F[B]                                        =
       FG.bracket(fa)(use) { case (a, _) => release(a) }
 
+    /**
+      * Guarantee finalization of [[fa]]. `release` is called in case of error
+      * @param release function to release value contained in [[fa]]
+      * @return `F[B]` updated value
+      */
     def guaranteeIncomplete[B](release: F[B])(implicit F: Applicative[F], FG: Guarantee[F]): F[A] =
       FG.bracket(F.unit)(_ => fa)((_, success) => release.whenA(!success))
 
+    /**
+      * Guarantee finalization of [[fa]]. `release` is alwyas called
+      * @param release function to release value contained in [[fa]]
+      * @return `F[A]` updated value
+      */
     def guaranteeAlways[B](release: F[B])(implicit F: Applicative[F], FG: Guarantee[F]): F[A] =
       FG.bracket(F.unit)(_ => fa)((_, _) => release)
 
+    /**
+      * Guarantee finalization of [[fa]].
+      * `release` is always called and boolean indicating success of usage is passed to it
+      * @param release function to release value contained in [[fa]] depending on success of usage
+      * @return `F[A]` updated value
+      */
+    def guaranteeOpt[B](release: Boolean => F[B])(implicit F: Applicative[F], FG: Guarantee[F]): F[A] =
+      FG.bracket(F.unit)(_ => fa)((_, success) => release(success))
+
+    /**
+      * Update value in [[fa]]. In case of error old value of [[fa]] is passed to `commit`, otherwise
```

Comment:
Well, unsuccessful result is not only the error case, but could be also a cancellation.
I suppose it should be rephrased